graphql2/graphqlapp/alert.go

```diff
@@ -39,6 +41,36 @@ func (a *AlertLogEntry) Message(ctx context.Context, obj *alertlog.Entry) (strin
 	return e.String(), nil
 }
 
+func (a *AlertLogEntry) State(ctx context.Context, obj *alertlog.Entry) (*graphql2.AlertLogEntryState, error) {
+	e := *obj
+	meta, ok := e.Meta().(*alertlog.NotificationMetaData)
+	if !ok {
+		return nil, nil
+	}
+
+	s, err := (*App)(a).FindOneAlertLogMessageStatus(ctx, meta.MessageID)
+	if err != nil {
+		return nil, errors.Wrap(err, "find alert log state")
+	}
+
+	status := ""
+	switch s.State {
+	case notification.MessageStateFailedPerm:
+		status = "ERROR"
+	case notification.MessageStateDelivered:
```

Comment:
Sent should probably be OK as well -- we don't always have delivery confirmation.
```suggestion
	case MessageStateSent, notification.MessageStateDelivered:
```

Reply:
Should we then change the UI to say "Sent" when sent? I don't think it would make sense to display "Delivered" on the UI when it might not have

Reply:
Right now, the UI displays "Delivered" if the status is "OK"

Reply:
Edit: See below, this will have to remain out of scope for more granular details wording